components/Newsfeed/Newsfeed.tsx

```diff
@@ -0,0 +1,276 @@
+    } else {
+      trolley ? setOrgFilter(false) : setBillFilter(false)
+    }
+  }
+
+  const checkDisplay = useCallback(() => {
+    setNoDisplay(false)
+    if (orgFilter == true && billFilter == true) {
+      return setNotificationsDisplay(allResults)
+    } else if (orgFilter == true) {
+      return setNotificationsDisplay(orgResults)
+    } else if (billFilter == true) {
+      return setNotificationsDisplay(billResults)
+    }
+    return setNoDisplay(true)
+  }, [allResults, orgFilter, orgResults, billFilter, billResults])
+
+  const notificationQuery = async () => {
+    let notificationList: Props = []
+    const q = query(subscriptionRef, where("uid", "==", `${uid}`))
+    const querySnapshot = await getDocs(q)
+    querySnapshot.forEach(doc => {
+      // doc.data() is never undefined for query doc snapshots
+      notificationList.push(doc.data().notification)
+    })
+
+    if (notificationsDisplay.length === 0 && notificationList.length != 0) {
+      setAllResults(notificationList)
+      setOrgResults(
+        notificationList.filter(notification => notification.type === "org")
+      )
+      setBillResults(
+        notificationList.filter(notification => notification.type === "bill")
+      )
+      setNotificationsDisplay(allResults)
+    }
+  }
+
+  useEffect(() => {
+    uid ? notificationQuery() : null
+  })
+
+  useEffect(() => {
+    checkDisplay()
+  }, [checkDisplay, orgFilter, billFilter, notificationsDisplay])
+
+  function Filters() {
+    return (
+      <FilterBoxs
+        isMobile={isMobile}
+        onFilterChange={onFilterChange}
+        orgFilter={orgFilter}
+        billFilter={billFilter}
+      />
+    )
+  }
+
+  return (
+    <>
+      {loading ? (
+        <Row>
+          <Spinner animation="border" className="mx-auto" />
+        </Row>
+      ) : (
+        <>
+          {profile ? (
+            <>
+              <StyledContainer>
+                <Header>
+                  <HeaderTitle>Newsfeed</HeaderTitle>
+                  {isMobile ? (
+                    <Filters />
+                  ) : (
+                    <Col className="d-flex flex-column">
+                      <Filters />
+                    </Col>
+                  )}
+                </Header>
+                {noDisplay ? (
+                  <div className="pb-4">
+                    <AlertCard
+                      header={`No Results`}
+                      subheader={``}
+                      timestamp={``}
+                      headerImgSrc={``}
+                      bodyImgSrc={``}
+                      bodyImgAltTxt={``}
+                      bodyText={`There are no news updates for your current followed topics`}
+                    />
+                  </div>
+                ) : (
+                  <>
+                    {notificationsDisplay.map(
+                      (element: ElementProps, index: number) => (
+                        <div className="pb-4" key={index}>
```

Comment:
```suggestion
                      (element: ElementProps) => (
                        <div className="pb-4" key={element.id}>
```
See [pitfalls section of docs on `key`](https://react.dev/learn/rendering-lists#where-to-get-your-key)

Reply:
done